docs/network-policy.md

```diff
@@ -0,0 +1,70 @@
+# NetworkPolicy
+Kubernetes NetworkPolicy documentation: https://kubernetes.io/docs/concepts/services-networking/network-policies/#behavior-of-to-and-from-selectors
+
+Each NetworkPolicy object consists of three sections:
+
+1. `podSelector`: a label selector that determines which pods the NetworkPolicy applies to
+2. `ingress`: determines which entities ingress traffic can be received from by the pods selected by the NetworkPolicy
+3. `egress`: determines which entities egress traffic can be sent to by pods selected by the NetworkPolicy
```

Comment:
I would remove the words "ingress" and "egress" and just say "traffic". Obviously the pods can't receive egress traffic or send ingress traffic. (And I misread it the first time, assuming there was a missing apostrophe, and it was saying "determine (which entities' ingress traffic) can be received from by the pods" and was like "you mean 'which entities' egress traffic can be received by the pods'.)

Also, possibly "selected by the `podSelector`" rather than "selected by the NetworkPolicy". It may be worth distinguishing `spec.podSelector` or "the toplevel `podSelector`" from `podSelector`s inside the `ingress` and `egress` sections.

This also ignores `policyTypes` and the asymmetry between ingress and egress there, which we can't really ignore if you want to explain default-deny correctly below...

Reply:
could this section be cleared up by showing a sample network policy and explaining the ingress and egress sections?